students/Michael Hrstka/Lab 4/turn_signal/turn_signal.c

```diff
@@ -0,0 +1,142 @@
+#include <avr/sfr_defs.h>
+// ------- Preamble -------- //
+#include <avr/io.h>
+#include <util/delay.h>
+
+#define clearRight()    PORTD &= 0b11110000;
+#define setRight(bit)   PORTD |= (1 << 3-bit);
+#define fullRight()     PORTD |= 0b00001111;
+#define clearLeft()     PORTD &= 0b00001111;
+#define setLeft(bit)    PORTD |= ((1 << bit)) << 4;
+#define fullLeft()      PORTD |= 0b11110000;
+
+int ledLPos = 0;
+int ledRPos = 0;
+int bitOff = 0;
+int maxLed = 3;
+int tickSpeed = 100;
+_Bool rightPressed = 0;
+_Bool leftPressed = 0;
+_Bool breakPressed = 0;
+int leftPin = 0;
+int rightPin = 2;
+int breakPin = 1;
+int hazardTimer = 0;
+_Bool hazardLit = 0;
+int hazardMaxTime = 4;
+_Bool rightDown = 0;
+_Bool leftDown = 0;
+_Bool breakDown = 0;
+_Bool hazardPressed = 0;
+
+void flashSignal() {
+
+  if(hazardPressed){
+    if(hazardLit) {
+      fullRight();
+      fullLeft();
+    } else {
+      clearRight();
+      clearLeft();
+    }
+    if(hazardTimer < hazardMaxTime) {
+      hazardTimer++;
+    } else {
+      hazardTimer = 0;
+      hazardLit = !hazardLit;
+    }
+  } else {
+    if(rightPressed || ledRPos > 0){
+      // Flash right signal
+      clearRight();
+      setRight(ledRPos)
+      ledRPos = ledRPos < 3 ? ledRPos + 1 : 0;
+    } else {
+      ledRPos = 0;
+
+      if(breakPressed)
+        fullRight()
+      else
+        clearRight();
+    }
+    if(leftPressed || ledLPos > 0) {
+      // Flash left signal
+      clearLeft();
+      setLeft(ledLPos);
+      ledLPos = ledLPos < 3 ? ledLPos + 1 : 0;
+    } else {
+      ledLPos = 0;
+
+      if(breakPressed)
+        fullLeft()
+      else
+        clearLeft();
+    }
+  }
+}
+
+void checkBtns(){
```

Comment:
Oof this function is pretty logic-heavy. Could probably use some comments, but otherwise looks good. I like the use of variables to signal later action, rather than doing the thing right in the middle of the button check.